From: Enze Li <lienze@kylinos.cn>
Subject: mm/damon/ops-common: remove redundant local variable in damon_migrate_folio_list()
Date: Sat, 18 Jul 2026 08:21:25 +0800

In damon_migrate_folio_list(), we're redeclaring folio inside the first
while loop, but it just shadows the outer one.  Since the second loop uses
the outer folio anyway, the inner declaration is pointless.

Remove it to consistently reuse the same variable throughout the
function and improve readability.

Link: https://lore.kernel.org/20260718002125.637104-1-lienze@kylinos.cn
Signed-off-by: Enze Li <lienze@kylinos.cn>
Reviewed-by: SJ Park <sj@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/damon/ops-common.c |    2 --
 1 file changed, 2 deletions(-)

--- a/mm/damon/ops-common.c~mm-damon-ops-common-remove-redundant-local-variable-in-damon_migrate_folio_list
+++ a/mm/damon/ops-common.c
@@ -339,8 +339,6 @@ static unsigned int damon_migrate_folio_
 	LIST_HEAD(migrate_folios);
 
 	while (!list_empty(folio_list)) {
-		struct folio *folio;
-
 		cond_resched();
 
 		folio = lru_to_folio(folio_list);
_
